### PR TITLE
v1.13 Backports 2023-05-02

### PR DESCRIPTION
**Pull request by joamaki**
* [x] #24770 (@aditighag)
 * [x] #25033 (@giorio94)
 * [x] #25109 (@sayboras)
 * [x] #24979 (@giorio94)
 * [x] #25192 (@kaworu)
 * [x] #25174 (@joestringer)
 * [ ] #25171 (@ayesha-kr)
 * [x] #25189 (@brb)

PRs skipped due to conflicts:

 * #24905 (@gentoo-root)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
for pr in 24770 25033 25109 24979 25192 25174 25171 25189; do contrib/backporting/set-labels.py $pr done 1.13; done
```
or with
```
make add-labels BRANCH=v1.13 ISSUES=24770,25033,25109,24979,25192,25174,25171,25189
```

### CODE REVIEW

**Review by sayboras — APPROVED** [review]
Looks good for my changes ✅

**Comment by joamaki** [comment]
@gentoo-root skipped your PR as the egress SNAT tests didn't apply cleanly due to https://github.com/cilium/cilium/pull/24849 being only partially backported. Might be trivial to fix, but wasn't sure. Could you take a look at backporting it separately?

**Review by brb — APPROVED** [review]
My changes LGTM, thanks.

**Review by kaworu — APPROVED** [review]
Thanks @joamaki my patch LGTM.

File: install/kubernetes/cilium/values.yaml [hunk 1]
```diff
@@ -889,7 +889,7 @@ hubble:
         - sourceLabels:
             - __meta_kubernetes_pod_node_name
           targetLabel: node
-          replacement: ${1}
```
Comment (joamaki) [inline install/kubernetes/cilium/values.yaml]: Something probably broken on my local setup as this differs from what CI comes up with. Will try to figure out what's up.

Comment (joamaki) [reply install/kubernetes/cilium/values.yaml]: `envsubst -no-digit` fixes it. Seems like something changed in newer versions of `envsubst` and CI was probably using a very old version. I'll look into this issue in a separate PR and see what we can do about it.

**Review by giorio94 — APPROVED** [review]
My commits look good. Thanks!

**Review by joestringer — APPROVED** [review]
For #25174: :+1:

**Review by aditighag — APPROVED** [review]
Looks good for my commits. Thanks!

**Comment by joamaki** [comment]
/test-1.20-4.19

`test-1.20-4.19` seems to be [failing](https://jenkins.cilium.io/job/Cilium-PR-K8s-1.20-kernel-4.19/1920/testReport/junit/Suite-k8s-1/20/K8sDatapathConfig_Host_firewall_With_VXLAN/) similarly as the previous backport (fix being worked on in https://github.com/cilium/cilium/pull/25199). Rerunning it.

**Comment by joamaki** [comment]
/test-1.20-4.19

**Comment by joamaki** [comment]
/test-1.26-net-next

**Pull request merged**